src/main/java/com/fasterxml/jackson/core/JsonFactory.java

```diff
@@ -374,6 +384,8 @@ protected JsonFactory(JsonFactory src, ObjectCodec codec)
         _generatorDecorators = _copy(src._generatorDecorators);
         _streamReadConstraints = src._streamReadConstraints == null ?
             StreamReadConstraints.defaults() : src._streamReadConstraints;
+        _errorReportConfiguration = src._errorReportConfiguration == null ?
```

Comment:
Let's not allow `null`s, so remove check, just assign.

Actually should probably remove null checks from `_streamReadConstraints` case as well. We should never have those as `null` anywhere.